src/async-split-at.mjs

```diff
@@ -0,0 +1,69 @@
+import ensureAsyncIterable from './internal/ensure-async-iterable'
+import curry from './internal/curry'
+
+function asyncSplitAt (index, iterable) {
+  const iter = ensureAsyncIterable(iterable)[Symbol.asyncIterator]()
+  let position = 0
+  let firstCache = null
+  let firstCacheStart = null
+  let done = false
+
+  async function cleanUp () {
+    if (!done && typeof iter.return === 'function') {
+      await iter.return()
+    }
+  }
+
+  async function * first () {
+    try {
+      while (!done) {
+        if (firstCache) {
```

Comment:
It seems to me that you're suggesting that first needs to await before doing this check. That isn't the case. Second makes it synchronously when called.